server/src/main/java/io/deephaven/server/session/TicketResolver.java

```diff
@@ -123,6 +125,32 @@ <T> SessionState.ExportBuilder<T> publish(
     <T> SessionState.ExportBuilder<T> publish(
             SessionState session, Flight.FlightDescriptor descriptor, final String logId, @Nullable Runnable onPublish);
 
+    /**
+     * Publish the result of the source object as the result represented by the destination ticket.
+     *
+     * @param session the user session context
+     * @param ticket the ticket to publish to
+     * @param logId an end-user friendly identification of the ticket should an error occur
+     * @param onPublish an optional callback to invoke when the result is published
```

Comment:
I feel like the semantics of `onPublish()` are kind of weird now. In one case you call it after the export resolves, in the shared case you call it immediately (which begs the question, why use it?).